kore/src/Kore/Step/Condition/Evaluator.hs

```diff
@@ -33,7 +39,7 @@ simplify
     -- to predicates. Even better, delete this one and use Predicate.simplify.
 simplify predicate = do
     simplifiedPredicates <-
-        Branch.gather . simplifyPredicate
+        Branch.gather . Predicate.simplify 0
```

Comment:
Ok, so why is "Predicate.simplify" better?

Reply:
`Predicate.simplify` is recursive with a depth bound. `simplifyPredicate` calls `Predicate.simplify` and propagates the local bound. If `Kore.Step.Condition.Evaluator.simplify` is called inside `Predicate.simplify`, then the depth limit may prevent `simplifyPredicate` from simplifying any predicates at all! This starts over at depth 0. This is actually a revert of an older change which unintentionally changed this behavior.